src/delegate.erl

```diff
@@ -90,11 +90,23 @@ start_link(Name, Num) ->
     Name1 = delegate_name(Name, Num),
     gen_server2:start_link({local, Name1}, ?MODULE, [Name1], []).
 
+invoke(Pid, FunOrMFA = {gen_server2, _F, _A}) when is_pid(Pid) ->  %% optimisation
+    case safe_invoke(Pid, FunOrMFA) of
+        {ok,    _, Result} -> Result;
+        {error, _, {Class, Reason, StackTrace}}  -> 
+            erlang:raise(Class, Reason, StackTrace)
```

Comment:
We do not use `erlang:raise/3` (or `exit/1`, or `throw/1`) in this module. Are you sure that it is safe to just terminate the caller?

Reply:
Could I make the following changes and resubmit the PR?
1. Fork a new branch without any rejected changes and reverts
2. Change  "S_v"  to  "SinglePid"

Reply:
Sure. Do you have any contention and/or throughput or latency numbers to share?

Reply:
Also, the above comment about the use of `erlang:raise/3` should be addressed. This module's API at the moment does not throw errors.